pkg/proxy/creator/creator.go

```diff
@@ -0,0 +1,238 @@
+	if err != nil {
+		return nil, err
+	}
+
+	economicMetricsCacher := cache.NewGenericApiResponseMemoryCacher()
+	cacheValidity = time.Duration(args.Config.GeneralSettings.EconomicsMetricsCacheValidityDurationSec) * time.Second
+
+	nodeStatusProc, err := processProxy.NewNodeStatusProcessor(bp, economicMetricsCacher, cacheValidity)
+	if err != nil {
+		return nil, err
+	}
+
+	proxyInstance.closableComponents.Add(nodeGroupProc, valStatsProc, nodeStatusProc, bp)
+
+	//nodeGroupProc.StartCacheUpdate()
+	valStatsProc.StartCacheUpdate()
+	nodeStatusProc.StartCacheUpdate()
+
+	blockProc, err := processProxy.NewBlockProcessor(connector, bp)
+	if err != nil {
+		return nil, err
+	}
+
+	blocksPrc, err := processProxy.NewBlocksProcessor(bp)
+	if err != nil {
+		return nil, err
+	}
+
+	proofProc, err := processProxy.NewProofProcessor(bp, pubKeyConverter)
+	if err != nil {
+		return nil, err
+	}
+
+	esdtSuppliesProc, err := processProxy.NewESDTSupplyProcessor(bp, scQueryProc)
+	if err != nil {
+		return nil, err
+	}
+
+	statusProc, err := processProxy.NewStatusProcessor(bp, statusMetricsProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	aboutInfoProc, err := processProxy.NewAboutProcessor(bp, common.UnVersionedAppString, common.UndefinedCommitString)
+	if err != nil {
+		return nil, err
+	}
+
+	facadeArgs := versionsFactory.FacadeArgs{
+		ActionsProcessor:             bp,
+		AccountProcessor:             accntProc,
+		FaucetProcessor:              faucetProc,
+		BlockProcessor:               blockProc,
+		BlocksProcessor:              blocksPrc,
+		NodeGroupProcessor:           nodeGroupProc,
+		NodeStatusProcessor:          nodeStatusProc,
+		ScQueryProcessor:             scQueryProc,
+		TransactionProcessor:         txProc,
+		ValidatorStatisticsProcessor: valStatsProc,
+		ProofProcessor:               proofProc,
+		PubKeyConverter:              pubKeyConverter,
+		ESDTSuppliesProcessor:        esdtSuppliesProc,
+		StatusProcessor:              statusProc,
+		AboutInfoProcessor:           aboutInfoProc,
+	}
+
+	apiConfigPath := path.Join(args.PathToConfig, "apiConfig")
+	apiConfigParser, err := versionsFactory.NewApiConfigParser(apiConfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	versionsRegistry, err := versionsFactory.CreateVersionsRegistry(facadeArgs, apiConfigParser)
+	if err != nil {
+		return nil, err
+	}
+	port := args.Config.GeneralSettings.ServerPort
+
+	proxyInstance.httpServer, err = api.CreateServer(
+		versionsRegistry,
+		port,
+		args.Config.ApiLogging,
+		config.CredentialsConfig{},
+		statusMetricsProvider,
+		args.Config.GeneralSettings.RateLimitWindowDurationSeconds,
+		false,
+		false,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return proxyInstance, nil
+}
+
+// Start will start the proxy
+func (p *proxy) Start() {
+	go func() {
+		err := p.httpServer.ListenAndServe()
+		if err != nil {
+			log.Error("cannot ListenAndServe()", "err", err)
+			os.Exit(1)
```

Comment:
return error here? there is already an `os.Exit(1)` in case of error

Reply:
removed and changed the log